File: programmers/pg_lv1_가장가까운글자.js
function solution(s) {
  let sArr = s.split("");
  let alphabetMap = new Map();
  let result = [];

  sArr.forEach((ele, idx) => {
    if (!alphabetMap.has(ele)) {
      result.push(-1);
      alphabetMap.set(ele, idx);
    } else {
      let prevIdx = alphabetMap.get(ele);
      result.push(idx - prevIdx);
      alphabetMap.set(ele, idx);
    }
  });

  return result;
}

File: python/programmers/pg_lv1_가장가까운글자.py
def solution(s):
    sList = list(s)
    alpDict = dict()
    result = []
    
    for (idx, i) in enumerate(sList):
        if i in alpDict: 
            alpIdx = alpDict.get(i)
            result.append(idx - alpIdx)
        else: result.append(-1)
        alpDict[i] = idx
        
    return result